### ISSUE TITLE
how set biopython for python2.7

### ISSUE THREAD

**Issue by biotech70** [biopython/biopython#3694]
Hi dear friends. I installed biopython on my system (ubuntu 20.04) and need it for work with some biological tools, but get Bio.pdb error. whats the causes and how can solve it? please guide me. Indeed my default python is python3 and biopython was installed for it but I need to python2.7 and biopython for it. how can I shift from biopython3 to biopython for python2.7?
![1](https://user-images.githubusercontent.com/73783725/129455303-c9d8f4f5-a05e-486e-9c8d-4c4be53aeb6c.png)
 shift to python2.7 and compile biopython for it?

**Comment by peterjc** [comment]
Quoting our README file, "Biopython 1.76 was our final release to support Python 2.7", so if you really must use Python 2.7, then use Biopython 1.76. However Python 2.7 is not supported anymore by the Python team. See e.g. https://www.python.org/doc/sunset-python-2/

It is not clear why you think you need to use Python 2.7. What tool are you trying to use?

**Comment by biotech70** [comment]
thanks for your response. I need python 2.7 and a compatible biopyhton version for it to run the ROSETTA package. when I used python 3.8 get the below error.
[error.txt](https://github.com/biopython/biopython/files/6987775/error.txt)
![1](https://user-images.githubusercontent.com/73783725/129472319-fa4348a6-44ae-472c-9df5-41d123ed6978.png)

**Comment by biotech70** [comment]
in addition, I send occurred errors during work with the desired script along with python2.7 and python3.8.
![2](https://user-images.githubusercontent.com/73783725/129472490-a25085a3-2c2a-4f09-b88f-ea0185f3d1c7.png)

**Comment by JoaoRodrigues** [comment]
When you use apt to install packages, they will install to the system
python. From your screenshot, you have anaconda/miniconda installed. Try
doing "python -m pip install biopython". That will install biopython to the
conda environment you have active.

**Comment by biotech70** [comment]
thanks for your solution I do it and will tell you the results.

**Comment by biotech70** [comment]
I try your solution but not solved my problem.
![Untitled](https://user-images.githubusercontent.com/73783725/129483752-e43aeb2d-c368-47fe-87b4-5ac2674604a5.png)

**Comment by peterjc** [comment]
The message ``SyntaxError: Missing parenthesis in call to 'print'.`` happens under Python 3 if using code written expecting Python 2.

Which "ROSETTA" package are you trying to use? URL please - there are lots of projects using that name.

**Comment by peterjc** [comment]
And to expand on Joao's suggestion: ``python -m pip install biopython`` will install for the default version of Python. That appears to be Python 3 on your machine?

If you really want to use Python 2, try ``python2.7 -m pip install biopython==1.76`` instead.

**Comment by JoaoRodrigues** [comment]
The Rosetta here is the structure prediction Rosetta, from the Baker lab.

Apparently it needs python2 to run.. what I'd do in your place is to create
a conda environment for Python 2, activate it, install biopython there. Not
sure pip will be able to resolve which version is still compatible, you
might need to grab it from our release page on GitHub and install it
manually.

General tip of using conda: never ever install anything on the base
environment, always use specific environments.

Tldr

conda create --name rosetta_env python=2.7
conda activate rosetta_env
python -m pip install biopython
python ...../rosetta/whatever.py

A domingo, 15/08/2021, 12:55, Peter Cock ***@***.***>
escreveu:

**Comment by biotech70** [comment]
Dear Rodrigues problem solved. Thanks for your guidance

**Comment by peterjc** [comment]
Excellent, I'm going to close this issue now then.
